elephant/test/test_spike_train_generation.py

```diff
@@ -69,6 +68,66 @@ def test_threshold_detection(self):
         except AttributeError: # If numpy version too old to have allclose
             self.assertTrue(np.array_equal(spike_train,spike_train))
 
+
+class AnalogSignalPeakDetectionTestCase(unittest.TestCase):
+
+    def setUp(self):
+        curr_dir = os.path.dirname(os.path.realpath(__file__))
+        npz_file_loc = os.path.join(curr_dir, 'spike_extraction_test_data.npz')
+        iom2 = neo.io.PyNNNumpyIO(npz_file_loc)
+        data = iom2.read()
+        self.vm = data[0].segments[0].analogsignals[0]
+        self.true_time_stamps = [0.0124,  0.0354,  0.0713,  0.1192,  0.1695,
+                                 0.2201,  0.2711] * second
+
+    def test_peak_detection_time_stamps(self):
+        # Test with default arguments
+        result = stgen.peak_detection(self.vm)
+        self.assertEqual(len(self.true_time_stamps), len(result))
+        self.assertIsInstance(result, neo.core.SpikeTrain)
+
+        try:
```

Comment:
In `AnalogSignalThresholdDetectionTestCase` it is commented that the try: except AttributeError: is supposed to catch the case `#If` numpy version too old to have `allclose` but from looking at the source code, `assert_array_almost_equal` does not use `allclose` in numpy v1.12.0. Even if it did, it would be a major bug if assert_array_almost_equal would use a not implemented function in any numpy version.